IntegrationTests/src/bkr/inttest/labcontroller/test_watchdog.py

```diff
@@ -317,6 +317,77 @@ def test_handles_non_ascii_characters_in_openstack_console(self, test_get_consol
+            self.recipe = data_setup.create_recipe()
+            self.guest_recipe = data_setup.create_guestrecipe(self.recipe)
+            job = data_setup.create_job_for_recipes([self.recipe, self.guest_recipe])
+            self.addCleanup(self.cleanup_job, job)
+
+            data_setup.mark_recipe_running(self.recipe, system=self.system)
+            data_setup.mark_recipe_installing(self.guest_recipe, system=self.system)
+
+            self.console_log = os.path.join(get_conf().get('CONSOLE_LOGS'), self.system.fqdn)
+            self.cached_console_log = os.path.join(get_conf().get('CACHEPATH'),
+                                                   'recipes',
+                                                   str(self.recipe.id // 1000) + '+',
+                                                   str(self.recipe.id), 'console.log')
+
+    def tearDown(self):
+        # Clear file created beneath test-consoles subdirectory.
+        if self.console_log and os.path.isfile(self.console_log):
+            os.remove(self.console_log)
+
+    def test_stores_guest_console_logs(self):
+
+        # Verify Guest recipe handled well and host logfile present.
+
+        first_line = 'Here is the first line of the log file.\n'
+        open(self.console_log, 'w').write(first_line)
+
+        self.watchdog = Watchdog()
+
+        self.monitor = Monitor({'recipe_id': self.recipe.id, 'is_virt_recipe': False,
+                               'system': self.system.fqdn}, self.watchdog)
+        self.assert_(self.monitor.console_watch.logfiles)
+
+        self.monitor_guest = Monitor({'recipe_id': self.guest_recipe.id, 'is_virt_recipe': False,
+                               'system': None}, self.watchdog)
+
+        self.assert_(not self.monitor_guest.console_watch.logfiles)
+
+        wait_for_condition(self.check_console_log_registered)
+        wait_for_condition(lambda: self.check_cached_log_contents(first_line))
+
+    def test_spawn_monitor_exception(self):
+
+        # Verify self.watchdog.pool() handles exception raised by spawn_monitor().
+        # Logs message without exception that there was a failure.
+        # Moves onto the next recipe.  This is all done with call to poll().
+
+        first_line = 'Here is the first line of the log file.\n'
+        open(self.console_log, 'w').write(first_line)
+        self.watchdog = Watchdog()
+
+        self.monitor = Monitor({'recipe_id': self.recipe.id, 'is_virt_recipe': False,
+                               'system': self.system.fqdn}, self.watchdog)
+        self.assert_(self.monitor.console_watch.logfiles)
+
+        self.monitor_guest = Monitor({'recipe_id': self.guest_recipe.id, 'is_virt_recipe': False,
+                               'system': None}, self.watchdog)
+        self.assert_(not self.monitor_guest.console_watch.logfiles)
+
+        with patch.object(Watchdog, 'spawn_monitor', side_effect=TypeError('Boom!')):
```

Comment:
You patching it here, I would expect to see some verification afterwards.

Reply:
The verification was that it didn't crash and the log entry I wrote didn't crash.  Code has since been removed.